js/actions/index.ts

```diff
@@ -51,15 +53,34 @@ export const API_CALL = "API_CALL";
 // REQUEST_PORTAL_DATA action will be processed by the reducer immediately.
 // See: api-middleware.js
 export function fetchAndObserveData() {
-  return {
-    type: REQUEST_PORTAL_DATA,
-    // Start with fetching portal data. It will cause other actions to be chained later.
-    callAPI: {
-      type: API_FETCH_PORTAL_DATA_AND_AUTH_FIRESTORE,
-      successAction: receivePortalData,
-      errorAction: fetchError,
-    },
-  };
+  const runKeyValue = queryValue("runKey");
+  if (runKeyValue) {
+    const activity= queryValue("activity") || "";
+    const source = activity? ((activity.split('/activities'))[0]).replace("https://","") : "";
+    const resourceUrl = queryValue("activity") || "";
```

Comment:
Seems like a duplicate of `activity` variable.

Reply:
Good catch!